src/Product/Importer.php

```diff
@@ -354,4 +357,27 @@ public function reconcile(array $identifiersToReconcileWith): void
             $this->dispatchPostEvent($product, 'update');
         }
     }
+
+    private function disableOldParentProductIfItHasNotAnyVariants(string $identifier, ProductInterface $product): void
+    {
+        $oldParentProduct = $this->productRepository->findOneByCode($identifier);
+        if ($oldParentProduct === null) {
+            return;
+        }
+        if ($oldParentProduct === $product) {
+            return;
+        }
```

Comment:
I'd have put this check in the import method as a wrapper for this method call. Something like this:
```
if ($oldParentProduct !== $product) {
    $this->disableOldParentProductIfItHasNotAnyVariants($identifier);
}
```
This way we'd have one less parameter and we call the method only if "necessary".

Reply:
We can remove the $product parameter but then we need to add the $oldParentProduct to check for the variants in it.

Reply:
Yes sorry, the $oldParentProduct variable is defined in this method.
We could do a check on the identifier/code instead
```
if ($identifier !== $product->getoCode()) {
    $this->disableOldParentProductIfItHasNotAnyVariants($identifier);
}
```
but only if it's not strictly necessary to do a check on the entire object ($oldParentProduct !== $product).

Reply:
Yes, it should works https://github.com/webgriffe/SyliusAkeneoPlugin/commit/6a97504527f94b678f991667ea621c536f9052b9. Done!

Reply:
Great!